src/scytale/primaryHandler.go

```diff
@@ -310,3 +274,48 @@ func NewPrimaryHandler(logger log.Logger, v *viper.Viper, registry xmetrics.Regi
 
 	return router, nil
 }
+
+func doFanout(ctx context.Context, fanout *http.Request, message *wrp.Message) (context.Context, error) {
+	populateMessage(ctx, message)
+
+	bookkeepingLog(ctx, message)
+
+	var buffer bytes.Buffer
+	if err := wrp.NewEncoder(&buffer, wrp.Msgpack).Encode(message); err != nil {
+		return ctx, err
+	}
+
+	fanoutBody := buffer.Bytes()
+	fanout.Body, fanout.GetBody = xhttp.NewRewindBytes(fanoutBody)
+	fanout.ContentLength = int64(len(fanoutBody))
+	fanout.Header.Set("Content-Type", wrp.Msgpack.ContentType())
+	fanout.Header.Set("X-Webpa-Device-Name", message.Destination)
+	return ctx, nil
+}
+
+func bookkeepingLog(ctx context.Context, message *wrp.Message) error {
```

Comment:
This function should not exist in application code.  What you want is decoration:

```
package logginghttp

import (
    "github.com/Comcast/webpa-common/logging"
    "net/http"
)

func Bookkeeper(next http.Handler) http.Handler {
    return http.HandlerFunc(func(response http.ResponseWriter, request *http.Request) {
        // rely on another decorator, e.g. xcontext.Populate, to supply the fields
        logging.GetLogger(request.Context()).Log(logging.MessageKey(), "Bookkeeping")
        next.ServeHTTP(response, request)
    })
}
```

Reply:
Client code, such as in `scytale`, will then supply a contextual logger, again via decoration:

```
// imports omitted
c := xcontext.Populate(0,
    logginghttp.SetLogger(loggerFromViper, logginghttp.Header("X-Example", "example")),
)

// use 'c' to decorate handlers:
alice.New(c, /* other constructors).Then(myHandler)
```